### PR TITLE
add HA dedupe functionality to docker dev environment

### PR DESCRIPTION
**Pull request by aldernero**
#### What this PR does

This PR updates the microservers docker compose development environment so that it can more easily be used to test the functionality and changes to the HA tracker. Specifically it
- adds an option `enable_ha_tracker` to toggle the functionality
- when the above is enabled, 2 additional prometheii are started that scrape the `memcache-exporter` and `load-generator` and add the relevant cluster and replica labels required by the HA tracker.
- the two targets scraped for the HA tracker are now removed from the base prometheus scrape targets
- the relevant flags to enable HA tracking were added to the mimir config

I chose to use 2 smaller prometheii instead of having 2 prometheii that scrape all the targets to save on memory requirements.

#### Checklist

- [N/A] Tests updated.
- [N/A] Documentation added.
- [x] `CHANGELOG.md` updated - the order of entries should be `[CHANGE]`, `[FEATURE]`, `[ENHANCEMENT]`, `[BUGFIX]`.
- [N/A] [`about-versioning.md`](https://github.com/grafana/mimir/blob/main/docs/sources/mimir/configure/about-versioning.md) updated with experimental features.

### CODE REVIEW

File: development/mimir-microservices-mode/docker-compose.jsonnet [hunk 1]
```diff
@@ -62,12 +64,14 @@ std.manifestYamlDoc({
       name: 'distributor-1',
       target: 'distributor',
       httpPort: 8000,
+      extraArguments: '-distributor.ha-tracker.consul.hostname=consul:8500',
```
Comment (56quarters) [inline development/mimir-microservices-mode/docker-compose.jsonnet:67]: This seems like it should be conditional based on the `enable_ha_tracker` config as well as the other HA config bits in `mimir.yaml`.

File: development/mimir-microservices-mode/docker-compose.jsonnet [hunk 2]
```diff
@@ -33,6 +33,7 @@ std.manifestYamlDoc({
 
     // If true, a query-tee instance with a single backend is started.
     enable_query_tee: false,
+    enable_ha_tracker: true,
```
Comment (56quarters) [inline development/mimir-microservices-mode/docker-compose.jsonnet:36]: Can you include some docs here (some of the info from the description of this PR would be good)?

File: development/mimir-microservices-mode/docker-compose.jsonnet (hunk 1 above)
Comment (56quarters) [reply development/mimir-microservices-mode/docker-compose.jsonnet:67]: Or alternatively, the jsonnet config should be removed so HA is always used.

**Review by 56quarters — APPROVED** [review]

File: development/mimir-microservices-mode/docker-compose.jsonnet (hunk 1 above)
Comment (aldernero) [reply development/mimir-microservices-mode/docker-compose.jsonnet:68]: I went with option 2, always have HA used and group the 2 new prometheus with existing `enable_prometheus` toggle

**Pull request merged**